app/components/primer/alpha/text_field.rb

```diff
@@ -0,0 +1,116 @@
+      #       name: :first_name,
+      #       label: "First name",
+      #       leading_visual: {
+      #         icon: :person
+      #       }
+      #     )
+      #   ) %>
+      #
+      # @example With a caption
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       caption: "What your friends call you"
+      #     )
+      #   ) %>
+      #
+      # @example With a validation message
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       validation_message: "Hmm, that doesn't look right"
+      #     )
+      #   ) %>
+      #
+      # @param name [String] Value for the HTML name attribute.
+      # @param id [String] Value for the HTML id attribute.
+      # @param class [String] A list of CSS classes to add to the input. Exists for compatibility with Rails form builders.
+      # @param classes [String] A list of CSS classes to add to the input. Combined with the `:class` argument.
+      # @param caption [String] Caption text to render below the input.
+      # @param label [String] Label text displayed above the input.
+      # @param show_label [Boolean] Whether or not to visually display the label. If `true` the label will still be available to screen readers.
+      # @param trailing_label [String] Text to display to the right of the input.
+      # @param size [Symbol] The size of the field. <%= one_of(Primer::Forms::Dsl::Input::SIZE_OPTIONS) %>
+      # @param show_clear_button [Boolean] Whether or not to include a clear button inside the input that clears the input's contents when clicked.
+      # @param clear_button_id [String] The HTML id attribute of the clear button.
+      # @param full_width [Boolean] Controls whether or not the input takes the full width of its container.
+      # @param disabled [Boolean] Whether or not the input should accept keyboard and mouse input.
+      # @param invalid [Boolean] If `true`, renders the input in a visually invalid state.
+      # @param placeholder [String] Placeholder text.
+      # @param inset [Boolean] If `true`, renders the input in a visually inset state.
+      # @param monospace [Boolean] If `true`, uses a monospace font for the input field.
+      # @param leading_visual [Hash] Renders a leading visual icon before the text field's cursor. The hash will be passed to Primer's [Octicon component](https://primer.style/view-components/components/octicon).
+      # @param validation_message [String] A validation message to display beneath the input. Implicitly sets `invalid` to `true`.
+      # @param label_arguments [Hash] System arugments passed to the Rails builder's `#label` method. These arguments will appear as HTML attributes on the `<label>` tag.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      # @param block [Proc] Unused.
```

Comment:
Can we remove this since `full_width` replaced it?

Reply:
Oh this is a dumb Ruby thing, i.e. is referring to a Ruby block (`Proc` object) rather than a `display: ` value. Some form inputs like `RadioButtonGroupInput` take a Ruby block that lets you define constituent radio buttons, eg:

```ruby
my_form.radio_button_group(name: :foo) do |group| # <-- this thing between the do/end keywords is a block
  group.radio_button(label: "bar")
  group.radio_button(label: "baz")
end
```

The constructor defined for this component lives in form_components.rb and accepts a block that may or may not be provided depending on what the input accepts. The code in form_components.rb simply passes it along to the input. I tried excluding it from the docs but Yard complained it existed in the code but wasn't documented.